qutip/solver/countstat.py

```diff
@@ -0,0 +1,225 @@
+    `c_ops` corresponding to monitored currents, given the system
+    Liouvillian `L`. The current collapse operators `c_ops` should be part
+    of the dissipative processes in `L`, but the `c_ops` given here does not
+    necessarily need to be all collapse operators contributing to dissipation
+    in the Liouvillian. Optionally, the steadystate density matrix `rhoss`
+    and the current operators `J_ops` correpsonding to the current collapse
+    operators `c_ops` can also be specified. If either of
+    `rhoss` and `J_ops` are omitted, they will be computed internally.
+    'wlist' is an optional list of frequencies at which to evaluate the noise
+    spectrum.
+
+    Parameters
+    ----------
+
+    L : :class:`qutip.Qobj`
+        Qobj representing the system Liouvillian.
+
+    c_ops : array / list
+        List of current collapse operators.
+
+    rhoss : :class:`qutip.Qobj` (optional)
+        The steadystate density matrix corresponding the system Liouvillian
+        `L`.
+
+    wlist : array / list (optional)
+        List of frequencies at which to evaluate (if none are given, evaluates
+        at zero frequency)
+
+    J_ops : array / list (optional)
+        List of current superoperators.
+
+    sparse : bool [True]
+        Flag that indicates whether to use sparse or dense matrix methods when
+        computing the pseudo inverse. Default is false, as sparse solvers
+        can fail for small systems. For larger systems the sparse solvers
+        are recommended.
+
+    method : str, ['direct']
+        Method used to compute the noise. The default, 'direct' with
+        ``sparse=True``, compute the noise directly. It is the recommended
+        method for larger systems. Otherwise, the pseudo inverse is computed
+        using the given method. Pseudo inverse supports 'splu' and 'spilu' for
+        sparse matrices and 'direct', 'scipy' and 'numpy' methods for
+        ``sparse=False``.
+
+    .. note::
+        The algoryth is described in page 67 of "Electrons in nanostructures"
+        C. Flindt, PhD Thesis, available online:
+        https://orbit.dtu.dk/fedora/objects/orbit:82314/datastreams/file_4732600/content
+
+    Returns
+    --------
+    I, S : tuple of arrays
+        The currents `I` corresponding to each current collapse operator
+        `c_ops` (or, equivalently, each current superopeator `J_ops`) and the
+        zero-frequency cross-current correlation `S`.
+    """
+    if rhoss is None:
+        rhoss = steadystate(L, c_ops)
+
+    if J_ops is None:
+        J_ops = [sprepost(c, c.dag()) for c in c_ops]
+
+    if wlist is None:
+        wlist = [0.]
+
+    J_ops = [op.data for op in J_ops]
+
+    if not sparse or method != 'direct':
+        current, noise = _noise_pseudoinv(L, wlist, rhoss, J_ops,
+                                          sparse, method)
+    else:
+        rhoss_vec = operator_to_vector(rhoss).data
+        current, noise = _noise_direct(L, wlist, rhoss_vec, J_ops)
```

Comment:
```suggestion
    if sparse and method == 'direct':
        rhoss_vec = operator_to_vector(rhoss).data
        current, noise = _noise_direct(L, wlist, rhoss_vec, J_ops)
    else:
        current, noise = _noise_pseudoinv(L, wlist, rhoss, J_ops,
                                          sparse, method)
```
Removing the double negative makes the code much clearer for me.

Reply:
I would also pass `rhoss` and `J_ops` as Qobj directly and not data instances, since L is a Qobj. I also wouldn't apply `operator_to_vector` here and just apply it inside `_noise_direct` if needed. It's better if even internally function mostly take all Qobj or all data instances.